pkg/ipcache/ipcache.go

```diff
@@ -47,19 +47,39 @@ type Identity struct {
 	// a pod IP (other source) has the same IP.
 	shadowed bool
 
-	// createdFromMetadata indicates that this entry was created via the new
-	// metadata API. This is needed to know if it is safe to delete
-	// an IPCache entry when no further metadata is associated with its prefix.
+	// tainted indicates that this entry touched by the legacy Upsert API.
+	// This informs the metadata subsystem to retain the original source and
+	// to clean up the entry if the legacy API already dropped its reference.
+	// upsertLocked will ensure that this field remains true once set. In other
+	// words, there is no way to un-taint an entry.
 	// This field is intended to be removed once cilium/cilium#21142 has been
 	// fully implemented and all entries are created via the new metadata API
-	createdFromMetadata bool
+	tainted bool
```

Comment:
I'm not sure I like this naming; the synchronous API is still in-use today with endpoint IPs, and I don't expect that to change anytime soon. Likewise, FQDN-based CIDR identities are "expected" and thus not some sort of anomaly or degraded state.

Reply:
Sure, I can rename it to something that doesn't convey any value judgment!